internal/pkg/cli/storage_init.go

```diff
@@ -132,63 +251,252 @@ func (o *initStorageOpts) Ask() error {
+		dynamoTableNameValidation,
+		prompt.WithFinalMessage("Sort key:"),
+	)
+	if err != nil {
+		return fmt.Errorf("get DDB sort key: %w", err)
+	}
+	keyTypePrompt := fmt.Sprintf(fmtStorageInitDDBKeyTypePrompt, ddbKeyString)
+	keyTypeHelp := fmt.Sprintf(fmtStorageInitDDBKeyTypeHelp, ddbKeyString)
+
+	keyType, err := o.prompt.SelectOne(keyTypePrompt,
+		keyTypeHelp,
+		attributeTypesLong,
+		prompt.WithFinalMessage("Sort key datatype:"),
+	)
+	if err != nil {
+		return fmt.Errorf("get DDB sort key datatype: %w", err)
+	}
+	o.sortKey = key + ":" + keyType
+	return nil
+}
+func (o *initStorageOpts) askDynamoAttributes() error {
+	// Extra attributes have been specified by flags.
+	if len(o.attributes) != 0 {
+		return nil
+	}
+	// If --no-lsi has been specified, there is no need to ask for more attributes
+	if o.noLsi {
+		return nil
+	}
+
+	// If --no-sort has been specified, there is no need to ask for more attributes
+	if o.noSort {
+		return nil
+	}
+	attributeTypePrompt := fmt.Sprintf(fmtStorageInitDDBKeyTypePrompt, color.Emphasize(ddbAttributeString))
+	attributeTypeHelp := fmt.Sprintf(fmtStorageInitDDBKeyTypeHelp, ddbAttributeString)
+	for {
+		moreAtt, err := o.prompt.Confirm(storageInitDDBMoreAttributesPrompt, storageInitDDBMoreAttributesHelp, prompt.WithFinalMessage("Additional attributes?"))
+		if err != nil {
+			return fmt.Errorf("confirm add more attributes: %w", err)
+		}
+		if !moreAtt {
+			break
+		}
+		att, err := o.prompt.Get(storageInitDDBAttributePrompt,
+			storageInitDDBAttributeHelp,
+			dynamoTableNameValidation,
+			prompt.WithFinalMessage("Attribute:"),
+		)
+		if err != nil {
+			return fmt.Errorf("get DDB attribute name: %w", err)
+		}
+		attType, err := o.prompt.SelectOne(attributeTypePrompt,
+			attributeTypeHelp,
+			attributeTypesLong,
+			prompt.WithFinalMessage("Attribute type:"),
+		)
+		if err != nil {
+			return fmt.Errorf("get DDB attribute type: %w", err)
+		}
+
+		o.attributes = append(o.attributes, att+":"+attType)
+	}
+	return nil
+}
+
+func (o *initStorageOpts) askDynamoLSIConfig() error {
+	if o.noLsi {
+		return nil
+	}
+	if o.noSort {
+		return nil
+	}
+	if len(o.lsiSorts) != 0 {
+		return nil
+	}
+	addLsi, err := o.prompt.Confirm(
+		storageInitDDBLSIPrompt,
+		storageInitDDBLSIHelp,
+		prompt.WithFinalMessage("Alternate sort keys?"),
+	)
+	if err != nil {
+		return fmt.Errorf("confirm add LSI to table: %w", err)
+	}
+	o.noLsi = !addLsi
+	if !addLsi {
+		return nil
+	}
+
+	var attributeNames []string
+	for _, att := range o.attributes {
+		attributeParts, err := getAttrFromKey(att)
+		if err != nil {
+			return fmt.Errorf("get name from attribute: %w", err)
+		}
+		attributeNames = append(attributeNames, attributeParts.name)
+	}
+
+	names, err := o.prompt.MultiSelect(storageInitDDBLSINamePrompt,
+		storageInitDDBLSIHelp,
+		attributeNames,
+		prompt.WithFinalMessage("Alternate sort keys:"),
+	)
+	if err != nil {
+		return fmt.Errorf("get LSI sort keys: %w", err)
+	}
+	if len(names) > 5 {
```

Comment:
just a nit but maybe worth to make 5 a constant with the name like `maxAlternativeSortKeys`